packages/devtools_app/lib/src/screens/network/offline_network_data.dart

```diff
@@ -0,0 +1,69 @@
+// Copyright 2024 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:devtools_shared/devtools_shared.dart';
+
+import '../../shared/http/http_request_data.dart';
+import '../../shared/primitives/utils.dart';
+import 'network_model.dart';
+
+/// Class to encapsulate offline data for the [NetworkController].
+/// It is responsible for serializing and deserializing offline network data.
+class OfflineNetworkData with Serializable {
+  OfflineNetworkData({
+    required this.requests,
+    this.selectedRequestId,
+    this.recording = false,
+  });
+
+  /// Creates an instance of [OfflineNetworkData] from a JSON map.
+  factory OfflineNetworkData.fromJson(Map<String, dynamic> json) {
+    final List<dynamic> requestsJson = json['requests'] ?? [];
+    final requests = requestsJson
+        .map(
+          (e) => DartIOHttpRequestData.fromJson(
+            e as Map<String, dynamic>,
+            null,
+            null,
+          ),
+        )
+        .toList();
+
+    return OfflineNetworkData(
+      requests: requests,
+      selectedRequestId: json['selectedRequestId'] as String?,
+      recording: json['recording'] as bool? ?? false,
+    );
+  }
+  bool get isEmpty => requests.isNullOrEmpty;
+
+  /// List of current [DartIOHttpRequestData] network requests.
+  final List<DartIOHttpRequestData> requests;
+
+  /// Get a request by matching its `id` field.
+  // Temporarily added to check selection in the filtered requests data,
+  // until we have current requests data in place
+  NetworkRequest? getRequest(String id) {
+    // Search through the list of requests and return the one with the matching ID.
+    return requests.firstWhere(
+      (request) => request.id == id,
+    );
+  }
+
+  /// The ID of the currently selected request, if any.
+  final String? selectedRequestId;
+
+  /// Whether the recording state is enabled.
+  final bool recording;
```

Comment:
I don't think we need to track this field. Please remove.

Reply:
Yes I was about to remove this, realised it's not required